### ISSUE TITLE
Replaying a request causes querystring to repeat

### ISSUE THREAD

**Issue by detj** [visionmedia/superagent#763]
I want to craft a request once and then replay it a number of times in future. This works if you don't have any querystring parameters in the url. In case the original request contains qs, the qs repeats itself.

```js
let req = superagent.get('/items').query({ limit: 10, offset: 0 });

req.end(handle);    // this works  url is /items?limit=10&offset=0

// at some point in future

req.end(handle)    // url becomes /items?limit=10&offset=0&limit=10&offset=0 & doesn't work

function handle() {
    // handle response
}
```

It doesn't feel right to call `.end` on an already ended request too. But I didn't find any other API to replay a request. What do you think should be a solution to this?

**Comment by romanantl** [comment]
Yeah, it happen me too.

**Comment by pornel** [comment]
I'm not sure if that's something that was ever intended to work. There are lots of stateful properties within the request object that could go wrong like that.

You'll be better off writing a function that configures a new request from scratch each time (the cost of doing this is small).

**Comment by detj** [comment]
Okay.

**Comment by mattfysh** [comment]
1.8.3 + `.then` causes this also... req.then(fn).then(fn) results in the request being sent once per `.then`

**Comment by pornel** [comment]
`.then` is fixed in 2.0
